test/util/malicious/hasher.go

```diff
@@ -165,14 +210,53 @@ func (n *Hasher) HashLeaf(ndata []byte) ([]byte, error) {
 
 // MustHashLeaf is a wrapper around HashLeaf that panics if an error is
 // encountered. The ndata must be a valid leaf node.
-func (n *Hasher) MustHashLeaf(ndata []byte) []byte {
+func (n *NmtHasher) MustHashLeaf(ndata []byte) []byte {
 	res, err := n.HashLeaf(ndata)
 	if err != nil {
 		panic(err)
 	}
 	return res
 }
 
+// ValidateNodeFormat checks whether the supplied node conforms to the
+// namespaced hash format and returns ErrInvalidNodeLen if not.
+func (n *NmtHasher) ValidateNodeFormat(node []byte) (err error) {
+	expectedNodeLen := n.Size()
+	nodeLen := len(node)
+	if nodeLen != expectedNodeLen {
+		return fmt.Errorf("%w: got: %v, want %v", ErrInvalidNodeLen, nodeLen, expectedNodeLen)
+	}
+	return nil
+}
+
+// validateSiblingsNamespaceOrder checks whether left and right as two sibling
+// nodes in an NMT have correct namespace IDs relative to each other, more
+// specifically, the maximum namespace ID of the left sibling should not exceed
+// the minimum namespace ID of the right sibling. It returns ErrUnorderedSiblings error if the check fails.
```

Comment:
I see the comment above:

> The documentation of this file may be incorrect.

and I agree that minimizing the diff w/ upstream seems like a good idea. I think the docs for most of the file can remain intact but

[optional] since the implementation of this function has been modified from [upstream](https://github.com/celestiaorg/nmt/blob/48b8f96a8764c19ea89ca076986f844cb29e8e9d/hasher.go#L232-L251), proposal to do

```suggestion
// validateSiblingsNamespaceOrder is intentionally faulty in that it does not
// check that two siblings are ordered with respect to their namespaces.
```